include/aws/common/math.h

```diff
@@ -132,6 +165,31 @@ AWS_STATIC_IMPL bool aws_is_power_of_two(const size_t x);
  */
 AWS_STATIC_IMPL int aws_round_up_to_power_of_two(size_t n, size_t *result);
 
+AWS_STATIC_IMPL uint8_t aws_min_u8(uint8_t a, uint8_t b);
+AWS_STATIC_IMPL uint8_t aws_max_u8(uint8_t a, uint8_t b);
+AWS_STATIC_IMPL int8_t aws_min_i8(int8_t a, int8_t b);
+AWS_STATIC_IMPL int8_t aws_max_i8(int8_t a, int8_t b);
+AWS_STATIC_IMPL uint16_t aws_min_u16(uint16_t a, uint16_t b);
+AWS_STATIC_IMPL uint16_t aws_max_u16(uint16_t a, uint16_t b);
+AWS_STATIC_IMPL int16_t aws_min_i16(int16_t a, int16_t b);
+AWS_STATIC_IMPL int16_t aws_max_i16(int16_t a, int16_t b);
+AWS_STATIC_IMPL uint32_t aws_min_u32(uint32_t a, uint32_t b);
+AWS_STATIC_IMPL uint32_t aws_max_u32(uint32_t a, uint32_t b);
+AWS_STATIC_IMPL int32_t aws_min_i32(int32_t a, int32_t b);
+AWS_STATIC_IMPL int32_t aws_max_i32(int32_t a, int32_t b);
+AWS_STATIC_IMPL uint64_t aws_min_u64(uint64_t a, uint64_t b);
+AWS_STATIC_IMPL uint64_t aws_max_u64(uint64_t a, uint64_t b);
+AWS_STATIC_IMPL int64_t aws_min_i64(int64_t a, int64_t b);
+AWS_STATIC_IMPL int64_t aws_max_i64(int64_t a, int64_t b);
+AWS_STATIC_IMPL size_t aws_min_size(size_t a, size_t b);
+AWS_STATIC_IMPL size_t aws_max_size(size_t a, size_t b);
+AWS_STATIC_IMPL int aws_min_int(int a, int b);
+AWS_STATIC_IMPL int aws_max_int(int a, int b);
+AWS_STATIC_IMPL float aws_min_float(float a, float b);
+AWS_STATIC_IMPL float aws_max_float(float a, float b);
+AWS_STATIC_IMPL double aws_min_double(double a, double b);
+AWS_STATIC_IMPL double aws_max_double(double a, double b);
```

Comment:
Shouldn't these be named ```aws_int_max```/```aws_int_min``` etc, to match all the rest of our APIs?

Reply:
I was matching these:
```
aws_mul_u64_saturating()
aws_mul_u32_saturating()
aws_mul_size_saturating()
...
```